### PR TITLE
SAR-4103 - Created controller & updated redirects to terms controller…

### PR DESCRIPTION
**Pull request by AlexRimmerHMRC**
… & updated/added tests

### CODE REVIEW

File: it/uk/gov/hmrc/vatsignupfrontend/controllers/agent/CheckYourAnswersFinalControllerISpec.scala [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        "the company name service returned INTERNAL SERVER ERROR" should {
+          "return INTERNAL_SERVER_ERROR" in {
+            val model = SubscriptionRequestSummary(
+              vatNumber = testVatNumber,
+              businessEntity = LimitedCompany,
+              optNino = None,
+              optCompanyNumber = Some(testCompanyNumber),
+              optSautr = None,
+              optSignUpEmail = Some(testEmail),
+              transactionEmail = testEmail,
```
Comment (LZaharia) [inline it/uk/gov/hmrc/vatsignupfrontend/controllers/agent/CheckYourAnswersFinalControllerISpec.scala]: You could use the testAgentEmail constant in the tests

File: test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/CheckYourAnswersFinalControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.vatsignupfrontend.SessionKeys
+import uk.gov.hmrc.vatsignupfrontend.config.featureswitch.{DirectDebitTermsJourney, FinalCheckYourAnswer}
+import uk.gov.hmrc.vatsignupfrontend.config.mocks.MockControllerComponents
+import uk.gov.hmrc.vatsignupfrontend.connectors.mocks.MockSubscriptionRequestSummaryConnector
+import uk.gov.hmrc.vatsignupfrontend.helpers.TestConstants._
+import uk.gov.hmrc.vatsignupfrontend.httpparsers.SubscriptionRequestSummaryHttpParser.{SubscriptionRequestExistsButNotComplete, SubscriptionRequestUnexpectedError}
+import uk.gov.hmrc.vatsignupfrontend.models._
+import uk.gov.hmrc.vatsignupfrontend.models.companieshouse.{LimitedLiabilityPartnership, NonPartnershipEntity}
+import uk.gov.hmrc.vatsignupfrontend.services.mocks.{MockGetCompanyNameService, MockStoreVatNumberService, MockSubmissionService}
```
Comment (LZaharia) [inline test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/CheckYourAnswersFinalControllerSpec.scala]: There are some imports that are not being used

File: test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/SentClientEmailControllerSpec.scala [hunk 3]
```diff
@@ -61,12 +62,25 @@ class SentClientEmailControllerSpec extends UnitSpec with GuiceOneAppPerSuite wi
...
-        val result = TestSentClientEmailController.submit(testPostRequest.withSession(SessionKeys.emailKey -> testEmail))
-        status(result) shouldBe Status.SEE_OTHER
-        redirectLocation(result) shouldBe Some(routes.TermsController.show().url)
+          val result = TestSentClientEmailController.submit(testPostRequest.withSession(SessionKeys.emailKey -> testEmail))
+          status(result) shouldBe Status.SEE_OTHER
+          redirectLocation(result) shouldBe Some(routes.TermsController.show().url)
+        }
+      }
+      "redirect to the final check your answer page" when {
+        "the final check your answer feature switch is disabled" in {
```
Comment (LZaharia) [inline test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/SentClientEmailControllerSpec.scala]: Change test description - the FS is enaled

File: test/uk/gov/hmrc/vatsignupfrontend/views/agent/CheckYourAnswersFinalSpec.scala [hunk 4]
```diff
@@ -455,7 +455,7 @@ class CheckYourAnswersFinalSpec extends ViewSpec with SummarySectionTesting {
 
       testPage.shouldHaveAcceptAndSendButton()
 
-      "NOT have business entity and have rows for VRN, agent email, client email and contact preference" in {
+      "NOT have business entity and client email, but have rows for VRN, agent email, client email and contact preference" in {
```
Comment (LZaharia) [inline test/uk/gov/hmrc/vatsignupfrontend/views/agent/CheckYourAnswersFinalSpec.scala]: Update description of the test - it testing the client email id.

File: test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/CheckYourAnswersFinalControllerSpec.scala (hunk 2 above)
Comment (AlexRimmerHMRC) [reply test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/CheckYourAnswersFinalControllerSpec.scala]: Optimised imports

File: test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/SentClientEmailControllerSpec.scala (hunk 3 above)
Comment (AlexRimmerHMRC) [reply test/uk/gov/hmrc/vatsignupfrontend/controllers/agent/SentClientEmailControllerSpec.scala]: Changed description to be enabled

File: test/uk/gov/hmrc/vatsignupfrontend/views/agent/CheckYourAnswersFinalSpec.scala (hunk 4 above)
Comment (AlexRimmerHMRC) [reply test/uk/gov/hmrc/vatsignupfrontend/views/agent/CheckYourAnswersFinalSpec.scala]: Changed this and forgot about it, didn't need to be changed so reverted back to how it was

**Pull request merged**